components/leaderboard/LeaderboardFilter.tsx

```diff
@@ -0,0 +1,359 @@
+                    </FormItem>
+                  )}
+                />
+              )}
+            />
+
+            {/* Winrate Range Slider */}
+            <FormField
+              control={form.control}
+              name="minWinRate"
+              render={({
+                field: { value: minWinRate, onChange: onMinWinRateChange },
+                fieldState,
+              }) => (
+                <FormField
+                  control={form.control}
+                  name="maxWinRate"
+                  render={({
+                    field: { value: maxWinRate, onChange: onMaxWinRateChange },
+                  }) => (
+                    <FormItem>
+                      <FormLabel>Win rate</FormLabel>
+                      <Slider
+                        min={0}
+                        max={1}
+                        step={0.01}
+                        value={[minWinRate || 0, maxWinRate || 1]}
+                        onValueChange={(vals) => {
+                          onMinWinRateChange(vals[0]);
+                          onMaxWinRateChange(vals[1]);
+                        }}
+                        minStepsBetweenThumbs={1}
+                        className={inputChangedStyle(fieldState)}
+                      />
+                      <div className="flex justify-between text-sm text-muted-foreground">
+                        <Input
+                          type="number"
+                          value={minWinRate || 0}
+                          onChange={(e) =>
+                            onMinWinRateChange(Number(e.target.value))
+                          }
+                          min={0}
+                          max={maxWinRate || 1}
+                          step={0.01}
+                          className="w-20 p-1 text-center"
+                        />
+                        <Input
+                          type="number"
+                          value={maxWinRate || 1}
+                          onChange={(e) =>
+                            onMaxWinRateChange(Number(e.target.value))
+                          }
+                          min={minWinRate || 0}
+                          max={1}
+                          step={0.01}
+                          className="w-20 p-1 text-center"
+                        />
+                      </div>
+                    </FormItem>
+                  )}
+                />
+              )}
+            />
+
+            {/* Tiers */}
+            <FormField
+              control={form.control}
+              name="tiers"
+              render={({ field: { value, onChange }, fieldState }) => (
+                <FormItem>
+                  <FormLabel>Tiers</FormLabel>
+                  <MultipleSelect
+                    className={inputChangedStyle(fieldState)}
+                    selected={value || []}
+                    options={tierOptions}
+                    onChange={(values) => onChange(values)}
```

Comment:
```suggestion
                    onChange={onChange}
```
Almost positive this should work